components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -125,6 +127,43 @@ public OAuth2AuthorizeRespDTO authorize(OAuth2AuthorizeReqDTO oAuth2AuthorizeReq
         }
     }
 
+    /**
+     * Process the authorization request and issue an authorization code or access token depending
+     * on the Response Type available in the request.
+     *
+     * @param authzReqMsgCtx authzReqMsgCtx.
+     * @return <code>OAuth2AuthorizeRespDTO</code> instance containing the access token/authorization code
+     * or an error code.
+     */
+    public OAuth2AuthorizeRespDTO authorize(OAuthAuthzReqMessageContext authzReqMsgCtx) {
+        OAuth2AuthorizeReqDTO oAuth2AuthorizeReqDTO =  authzReqMsgCtx.getAuthorizationReqDTO();
+        if (log.isDebugEnabled()) {
+            log.debug("Authorization Request received for user : " + oAuth2AuthorizeReqDTO.getUser() +
+                    ", Client ID : " + oAuth2AuthorizeReqDTO.getConsumerKey() +
+                    ", Authorization Response Type : " + oAuth2AuthorizeReqDTO.getResponseType() +
+                    ", Requested callback URI : " + oAuth2AuthorizeReqDTO.getCallbackUrl() +
+                    ", Requested Scope : " + OAuth2Util.buildScopeString(
```

Comment:
Lets add requested and authorized scopes at this point

Reply:
fixed